templates/makefile-c++/include/README

```diff
@@ -0,0 +1,39 @@
+
+This directory is intended for project header files.
+
+A header file is a file containing C declarations and macro definitions
+to be shared between several project source files. You request the use of a
+header file in your project source file (C, C++, etc) located in `src` folder
+by including it, with the C preprocessing directive `#include'.
+
+```src/main.c
+
+#include "header.h"
```

Comment:
This  looks great, one style question: Do we want to have C++ headers with `.hpp` extensions to more easily differentiate from pure C?

Reply:
Think we should stick with `.h` headers.

Reply:
Fair enough, considering that we can run C++ on embedded too now we are probably ok not distinguishing file extensions.